autoload/ollama/edit.vim

```diff
@@ -5,10 +5,19 @@ let s:buf = -1
 " avoid starting a 2nd edit job while one is in progress
 let g:edit_in_progress = 0
 
+" Function to check if embedded Python is available
+function! ollama#edit#HasEmbeddedPython() abort
+    return exists('g:ollama_embedded_python') && g:ollama_embedded_python != 0
+endfunction
+
 " Define the VimScript callback function
 " This will be called from python when to operations is 'done'
 " or aborted with 'error'
 function! ollama#edit#EditCodeDone(status)
+    if !ollama#edit#HasEmbeddedPython()
+        echoerr "OllamaEdit features require Vim compiled with +python3 support."
```

Comment:
Considering the same error message is repeated after every invocation to `ollama#edit#HasEmbeddedPython()` maybe it would make sense to move the error message into the same function?

```vim
function! ollama#edit#EnsureEmbeddedPythonIsAvailable() abort
    if exists('g:ollama_embedded_python') && g:ollama_embedded_python != 0
        echoerr "OllamaEdit features require Vim compiled with +python3 support."
        return v:true
    else
        return v:false
    endif
endfunction
```

And then the usage looks like this:

```vim
    if !ollama#edit#EnsureEmbeddedPythonIsAvailable()
        return
    endif
```